apps/src/p5lab/spritelab/blocks.js

```diff
@@ -104,38 +105,52 @@ const customInputTypes = {
   locationPicker: {
     addInput(blockly, block, inputConfig, currentInputRow) {
       currentInputRow.appendField(
-        `${inputConfig.label}(0, 0)`,
+        `${inputConfig.label}`,
         `${inputConfig.name}_LABEL`
       );
       const fieldRow = currentInputRow.getFieldRow();
-      const label = fieldRow[fieldRow.length - 1];
+      const fieldLabel = fieldRow[fieldRow.length - 1]; // fieldLabel = 'sprite at'
       const icon = document.createElementNS(SVG_NS, 'tspan');
       icon.style.fontFamily = 'FontAwesome';
       icon.textContent = '\uf276';
-      const button = new Blockly.FieldButton(
-        icon,
-        updateValue => {
-          getLocation(loc => {
-            if (loc) {
-              button.setValue(JSON.stringify(loc));
-            }
-          });
-        },
-        block.getHexColour(), // Google Blockly includes block.getColour
-        value => {
-          if (value) {
-            try {
-              const loc = JSON.parse(value);
-              label.setValue(
-                `${inputConfig.label}(${loc.x}, ${APP_HEIGHT - loc.y})`
-              );
-            } catch (e) {
-              // Just ignore bad values
-            }
+      const onChange = () => {
+        getLocation(loc => {
+          if (loc) {
+            fieldButton.setValue(JSON.stringify(loc));
+          }
+        });
+      };
+      const onDisplaySetField = value => {
+        if (value) {
+          try {
+            const loc = JSON.parse(value);
+            return `(${loc.x}, ${APP_HEIGHT - loc.y})`;
+          } catch (e) {
+            // Just ignore bad values
           }
         }
+      };
+      // TODO: Remove function below when sprite lab migrates to google blockly.
+      const onDisplaySetLabel = value => {
```

Comment:
Would it make sense to move this to cdoBlocklyWrapper? I feel like doing so would align with the decision to include the custom validator in cdoUtils rather than here in blocks.js. Moving it to the wrapper would also mean we could remove the TODO, because it would naturally get removed once CDO Blockly is fully deprecated.